sdk/samples/iot/hub/src/paho_iot_hub_methods_example.c

```diff
@@ -308,79 +307,117 @@ static int subscribe()
   if ((rc = MQTTClient_subscribe(mqtt_client, AZ_IOT_HUB_CLIENT_METHODS_SUBSCRIBE_TOPIC, 1))
       != MQTTCLIENT_SUCCESS)
   {
-    printf("Failed to subscribe to method topic filter, return code %d\n", rc);
+    printf("Failed to subscribe to methods topic filter, return code %d\n", rc);
     return rc;
   }
 
-  return 0;
+  return rc;
 }
 
-int main()
+static int on_received(void* context, char* topicName, int topicLen, MQTTClient_message* message)
 {
-  int rc;
+  (void)context;
 
-  // Read in the necessary environment variables and initialize the az_iot_hub_client
-  if (az_failed(rc = read_configuration_and_init_client()))
+  printf("Received a message from service.\n");
+  printf("Topic: %s\n", topicName);
+
+  if (topicLen == 0)
   {
-    printf("Failed to read configuration from environment variables, return code %d\n", rc);
-    return rc;
+    // The length of the topic if there are one or more NULL characters embedded in topicName,
+    // otherwise topicLen is 0.
+    topicLen = (int)strlen(topicName);
   }
+  az_span topic_span = az_span_init((uint8_t*)topicName, topicLen);
 
-  // Get the MQTT client id used for the MQTT connection
-  size_t client_id_length;
+  // Parse the incoming message topic and check to make sure it is a methods message
+  az_iot_hub_client_method_request method_request;
   if (az_failed(
-          rc = az_iot_hub_client_get_client_id(
-              &client, mqtt_client_id, sizeof(mqtt_client_id), &client_id_length)))
+          az_iot_hub_client_methods_parse_received_topic(&client, topic_span, &method_request)))
   {
-    printf("Failed to get MQTT clientId, return code %d\n", rc);
-    return rc;
+    printf("Topic is not a methods message.\n");
   }
-
-  // Create the Paho MQTT client
-  if ((rc = MQTTClient_create(
-           &mqtt_client, mqtt_endpoint, mqtt_client_id, MQTTCLIENT_PERSISTENCE_NONE, NULL))
-      != MQTTCLIENT_SUCCESS)
+  else
   {
-    printf("Failed to create MQTT client, return code %d\n", rc);
-    return rc;
-  }
+    printf("Topic is a methods message.\n");
 
-  // Set the callback for incoming MQTT messages
-  if ((rc = MQTTClient_setCallbacks(mqtt_client, NULL, NULL, on_received, NULL))
-      != MQTTCLIENT_SUCCESS)
-  {
-    printf("Failed to set MQTT callbacks, return code %d\n", rc);
-    return rc;
+    if (az_span_is_content_equal(ping_method_name_span, method_request.name))
+    {
+      // Invoke Method
+      az_span response = ping_method();
+
+      // Send a response
+      int rc;
+      if ((rc = send_method_response(&method_request, 200, response)) != 0)
+      {
+        printf("Unable to send %d response, status %d\n", 200, rc);
+      }
+    }
+    else
+    {
+      // Unsupported Method
```

Comment:
printf("Method %.*s not found\n", az_span_len(method_request.name), az_span_ptr(method_request.name));